src/Text/Gigaparsec/Internal.hs

```diff
@@ -92,10 +108,20 @@ instance Monad Parsec where
 
 instance Alternative Parsec where
   empty :: Parsec a
-  empty = undefined -- TODO:
+  empty = Parsec $ \st _ err -> err st
 
   (<|>) :: Parsec a -> Parsec a -> Parsec a
-  (<|>) = undefined -- TODO:
+  Parsec p <|> Parsec q = Parsec $ \st ok err ->
+    let ok' x st'
```

Comment:
This is playing with fire a bit... It's true for now, but if we add something to the state later, we may not want to throw away those changes... I would suggest:

```suggestion
    let ok' x st' = ok x (st' { consumed = consumed st || consumed st' })
```

But I suppose that's allocating more than necessary, so I'm unconvinced...

Reply:
Actually, we can avoid retaining `st`, which people seem to favour avoiding space leaks.